lib/mercury/mercury.rb

```diff
@@ -44,10 +50,14 @@ def initialize(host: 'localhost',
   def publish(source_name, msg, tag: '', headers: {}, &k)
     # The amqp gem caches exchange objects, so it's fine to
     # redeclare the exchange every time we publish.
-    # TODO: wait for publish confirmations (@channel.on_ack)
     with_source(source_name) do |exchange|
-      exchange.publish(write(msg), **Mercury.publish_opts(tag, headers)) do
-        k.call
+      payload = write(msg)
+      pub_opts = Mercury.publish_opts(tag, headers)
+      if publisher_confirms_enabled
+        expect_publisher_confirm(k)
```

Comment:
Will waiting for publisher confirmations have any noticeable performance degradation? I can't think of any, since our rabbit messages are used for long-running calls where a few hundred milliseconds shouldn't make much difference.